lib/user.js

```diff
@@ -892,20 +895,40 @@ module.exports = function (config, userDB, couchAuthDB, mailer, emitter) {
         }
         user = result.rows[0].doc;
         hash = util.URLSafeUUID();
-        user.forgotPassword = {
-          token: hash,
-          issued: Date.now(),
-          expires: Date.now() + tokenLife * 1000
-        };
```

Comment:
This part is not completely DRY (don't repeat yourself). You can create issued and expired once and then add the token inside an if statement.